apps/parsley/src/hooks/useSections/index.ts

```diff
@@ -56,18 +65,24 @@ export const useSections = ({
 
   useEffect(() => {
     if (sectionData && sectionState === undefined) {
-      setSectionState(sectionData.reduce(closeAllSectionsReducer, {}));
+      setSectionState(populateSectionState(sectionData));
     }
   }, [sectionData, sectionState]);
 
-  const openSection = useCallback(
-    (functionName: string, isOpen: boolean) => {
-      if (sectionState) {
-        setSectionState((currentState) => ({
-          ...currentState,
-          [functionName]: { isOpen },
-        }));
-      }
+  const openSection: OpenSection = useCallback(
+    ({ commandID, functionID, isOpen }) => {
+      setSectionState((currentState) => {
+        if (currentState) {
+          const nextState = { ...currentState };
+          if (commandID) {
+            nextState[functionID].commands[commandID].isOpen = isOpen;
+          } else {
+            nextState[functionID].isOpen = isOpen;
+          }
```

Comment:
Is there any reason not to split out these functions into `openFunction` and `openCommand`? I feel like the logic is a little ambiguous (it's not super clear when `isOpen` refers to the function or command state) and may allow more flexibility in the future

Reply:
Good idea!